src/HexagonOptimize.cpp

```diff
@@ -679,6 +681,27 @@ class OptimizePatterns : public IRMutator {
                 expr = mutate(op->a + neg_b);
                 return;
             } else {
+                // No widening subtracts for the expressions of the form:
```

Comment:
Actually... before you go implement this with vmpa... I'm pretty sure this instruction already exists. I see a widening subtract for both u8 -> i16 and u16 -> i32 (as well as i16 -> i32).